docs/guides/build-args.md

```diff
@@ -0,0 +1,193 @@
+   ```
+
+2. From an environment variable
+
+   If no value is given for name, then earthly will look for the value in the corresponding
+   environment variable on the localhost:
+   
+   ```bash
+   export name="banana"
+   earthly --build-arg name +hello
+   ```
+
+3. From a `.env` file
+
+   It is also possible to create an `.env` file to contain the build arguments to pass
+   to earthly. First create an `.env` file with:
+   
+   ```
+   name eggplant
+   ```
+   
+   Then simply run earthly:
+   
+   ```bash
+   earthly +hello
+   ```
+
+## Passing Argument values to targets
+
+Build arguments can also be set when calling build targets. If multiple build arguments values are defined for the same argument name,
+earthly will build the target for each value. For example, we can create a new `greetings` target which calls `+hello` multiple times:
+
+```dockerfile
+greetings:
+    BUILD \
+        --build-arg name=world \
+        --build-arg name=banana \
+        --build-arg name=eggplant \
+        +hello
+```
+
+Then when we call `earthly +greetings`, earthly will call `+hello` three times:
+
+```
+     buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
+ alpine:latest | --> Load metadata linux/amd64
+         +base | --> FROM alpine:latest
+         +base | [██████████] resolve docker.io/library/alpine:latest@sha256:69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f ... 100%
+        +hello | name=banana
+        +hello | --> RUN echo "hello $name"
+        +hello | name=eggplant
+        +hello | --> RUN echo "hello $name"
+        +hello | name=world
+        +hello | --> RUN echo "hello $name"
+        +hello | hello banana
+        +hello | hello eggplant
+        +hello | hello world
+        output | --> exporting outputs
+```
+
+In addition to the `BUILD` command, the `--build-arg` flag can also be used with `FROM` and `COPY` commands.
+
+## Passing secrets to RUN commands
+
+Secrets are similar to build arguments; however, they are *not* defined in targets, but instead are explicitly defined for each `RUN` command that is permitted to access them.
+
+Here's an example Earthfile that accesses a secret stored under `+secrets/passwd` and exposes it under the environment variable `mypassword`:
+
+```dockerfile
+FROM alpine:latest
+hush:
+    RUN --secret mypassword=+secrets/passwd echo "my password is $mypassword"
+```
+
+{% hint style='info' %}
+It's also possible to temporarily mount a secret as a file:
+
+```dockerfile
+RUN --mount type=secret,target=/root/mypassword,id=+secrets/passwd echo "my password is $(cat /root/mypassword)"
+```
+
+The file will not be saved to the image snapshot.
+{% endhint %}
+
+The value for `+secrets/passwd` must then be supplied to earthly on the command line. Either directly via:
+
+```bash
+earthly --secret passwd=itsasecret +hush
+```
+
+or if the value is omitted, then earthly will attempt to lookup the value from an environment variable on the localhost:
```

Comment:
We should probably mention the option to store secrets in the cloud and link to the relevant guide.

EDIT: I see that it is mentioned below, but in a slightly different context. I would also mention here, without going into details.

Reply:
that's a good point, I added a small link.